pkg/controllers/pod_controller.go

```diff
@@ -116,6 +134,13 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		}
 	}
 
+	if r.replicaResourceProvider != nil {
```

Comment:
Will this be nil? Will we always have a default provider?

Reply:
No, it will not be nil. The replicaResourceProvider is initialized in `NewPodReconciler`, if we encounter an error when we initialize replicaResourceProvider, `NewPodReconciler` will return an error and exit the program, so here if initialization is successful, here will not be nil.

And I have kept a default empty implementation of defaultReplicaResourceProvider. If the kube-scheduler integrates gang-scheduling capabilities by default in the future, it will be used as the default implementation.